nbclient/util.py

```diff
@@ -44,14 +47,9 @@ def just_run(coro: Awaitable) -> Any:
     else:
         had_running_loop = True
     if had_running_loop:
-        # if there is a running loop, we patch using nest_asyncio
-        # to have reentrant event loops
-        check_ipython()
-        import nest_asyncio
-        nest_asyncio.apply()
-        check_patch_tornado()
-    return loop.run_until_complete(coro)
-
+        return pool.submit(asyncio.run, coro).result()
```

Comment:
We could have only this line in the function, so that we always run in a thread with its own event loop.